src/ucp/tag/tag_send.c

```diff
@@ -153,7 +153,8 @@ ucp_tag_send_inline(ucp_ep_h ep, const void *buffer, size_t count,
     length = ucp_contig_dt_length(datatype, count);
 
     if ((ssize_t)length <= ucp_ep_config(ep)->tag.max_eager_short &&
-        ucp_memory_type_cache_is_empty(ep->worker->context)) {
+        (ep->worker->context->memtype_cache &&
```

Comment:
it disables inline sends when mem_cache is of (even if we do not have any mem type mds), right? Why?

Reply:
Yes.  
Not sure if user will play with mem_cache on/off  option if environment not support mem type support(no mem type mds).

Reply:
you check the mem_cache itself, not the condig flag. If we have no mem type mds, mem_cashe will be NULL, right?
[mem_cache init](https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_context.c#L972)

Reply:
right. I missed that case.  I updated the PR,  Plz review.

Reply:
this seems to add extra branch, since before this change memtype_cache was off if no memtype MDs are present

Reply:
yes, it adds extra branches for fast path